### ISSUE TITLE
IPykernel 6.9.0 generates error when Python errors are returned after execution

### ISSUE THREAD

**Issue by ccordoba12** [ipython/ipykernel#860]
This is very simple to reproduce in QtConsole. After executing something like this:

![imagen](https://user-images.githubusercontent.com/365293/153250861-6c5a492c-20cc-4505-8d45-4616af9cba65.png)

you'll see the following traceback in the terminal where it was started

```python-traceback
Traceback (most recent call last):
  File "/home/carlos/miniconda3/envs/py310/lib/python3.10/site-packages/ipykernel/kernelbase.py", line 461, in dispatch_queue
    await self.process_one()
  File "/home/carlos/miniconda3/envs/py310/lib/python3.10/site-packages/ipykernel/kernelbase.py", line 450, in process_one
    await dispatch(*args)
TypeError: object NoneType can't be used in 'await' expression
```

I can't reproduce this in 6.8.0 or older versions and it's not related to the Python version I'm using (it was also reproduced in 3.8, as reported [here](https://github.com/spyder-ide/spyder/issues/17316)).

I'd say this was introduced in PR #853, but I don't know how to fix it.

**Comment by minrk** [comment]
Creating a test env to see if I can track it down, but 90% sure something still has `await self._abort_queues()` which is no longer a coroutine. Adding `async def` back would fix that, but I'd like to take a minute to understand it first.

**Comment by ccordoba12** [comment]
No problem. Thanks for taking a look at this @minrk!

**Comment by minrk** [comment]
I was close! Right issue, wrong function. Fixed by #861
